tlatools/org.lamport.tlatools/src/util/Resolver.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2025, Oracle and/or its affiliates.
+
+package util;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Optional;
+
+/**
+ * Resolves abstract filenames (e.g. <code>"Integers.tla"</code>) to concrete locations
+ * (e.g. <code>"jar:file:/path/to/tla2tools.jar!/tla2sany/StandardModules/Integers.tla"</code>).
+ *
+ * <p>This interface is simple enough to be composable: multiple resolvers can be chained together
+ * using {@link SequentialResolver}.
+ */
+interface Resolver {
+
+    /**
+     * Locate the given file.  The name is treated as a filename, not a TLA+ module name;
+     * if you are looking for the Integers module, pass <code>"Integers.tla"</code>.
+     *
+     * @param name the file to locate, e.g. <code>"Integers.tla"</code>
+     * @return a URL to the file, or {@link Optional#empty() empty} if it could not be found
+     * @throws IOException if an error occurs while searching
+     */
+    Optional<URL> resolve(String name) throws IOException;
```

Comment:
rename `name` to `filename`?

Reply:
The majority of the codebase uses `null` instead of `Optional` (except for two places outside of `test/` that escaped my review years ago). If you anticipate the `Resolver` interface to be used beyond `SimpleFilenameToStream`, I recommend aligning with the rest of the codebase.